src/plugins/vis_type_timeseries/public/application/components/vis_editor.tsx

```diff
@@ -117,76 +139,75 @@ export class VisEditor extends Component {
     this.setState({ dirty: false });
   };
 
-  handleAutoApplyToggle = (event) => {
+  handleAutoApplyToggle = (event: React.ChangeEvent<HTMLInputElement>) => {
     this.setState({ autoApply: event.target.checked });
   };
 
-  onDataChange = ({ visData }) => {
+  onDataChange = ({ visData }: { visData: TimeseriesVisData }) => {
     this.visDataSubject.next(visData);
   };
 
   render() {
-    const { model } = this.state;
-
-    if (model) {
```

Comment:
this check was simply replaced with `if (!visFields)`, since render is only waiting for fields fetched.